content/articles/securing-mongodb-in-linux/index.md

```diff
@@ -0,0 +1,230 @@
+### Introduction
+MongoDB is one of the most used modern web applications for a database. MongoDB is not secure by default. Once installed, one will have a difficult time before configuring it for authentication, since people can browse, write, delete, or modify data on your server without ever needing to log in or authenticate. Securing the database is simple and may be completed in a few steps. In this article, I'll show you how to protect your MongoDB setup by using some activation command lines.
+
+### Check the environment
+To check if MongoDB is running on Linux kernel, we use the command below:
+
+```bash
+#call mongod and specify the path.
+mongod --dbpath = "C:\data\mongo"
+#or use this below to check.
+ps -e|grep 'Mongod'
+```
+The output if MongoDB is installed will be similar to this:
+`909070 ttys023 2:43.486 ./mongodb`
+To configure MongoDB we navigate to the default file location `/etc/mongodb.conf` where it runs on the default port. To change this file location, we use the command below:
+
+```bash
+$sudo Andrew/local/etc/mongodb.conf
+```
+
+If MongoDB is installed on the same machine where your web app is installed, there may be some communication between the database with the application to end this kind of linkage set the Ip address to 127.0.0.1 this automatically ends communication via the internet.
+
+```bash
+#network 
+net:
+    port: 27274
+    bindIp: 127.0.0.1
+```
+
+Through binding Ip address this ensures that MongoDB runs in a safe network environment thus safeguarding incoming connection. This is done by changing the Ip address using the bind command:
+
+```bash
+#binding command for secure Vlan connection
+bindIp: 192.168.1.7
+```
+
+### Securing MongoDB
+To secure your MongoDB, we use the following steps:
+
+- Create a user with admin access.
+- Turn on authentication.
+- Assign administrative users access to the database.
+
+### Create a user with system access
+MongoDB version accepts connection from local Unix socket. As a result, it cannot connect a huge internet connection. Furthermore, authentication is disabled by default. Users having MongoDB installed can still have access to the database. To secure your database you need to add administrative user. To add an administrative user, we first connect to the Mongo  shell prompt using the command below: 
+
+```bash
+$mongo
+```
+
+The output displays the shell's version, the connection of MongoDB that is its IP address, and more warnings indicating database access control is not enabled. This shows that reading and writing to the database is unrestricted, insecure and anyone with access to the server can have control of the database. Due to this, we can check the list of databases available to the server. By running the command below:
+
+```bash
+#The command returns lists of every database on the server.
+show dbs
+```
+
+ something like this will appear:
+
+```output
+db-job-admin   0.0091GB
+db-jobs-new  0.0009GB
+db-config  0.0001GB
+local   0.0100GB
+```

```

Comment:
You cannot start a sentence with, "something....". Also, this is a `bash` command, not `output`. Go through the whole tutorial and edit accordingly.